File: test-projects/python-sample/utils/logger.py
"""
Logger utility module
"""

from enum import Enum
from datetime import datetime
from typing import Any


class LogLevel(Enum):
    """Log level enumeration"""
    DEBUG = 0
    INFO = 1
    WARN = 2
    ERROR = 3


class Logger:
    """Simple logger utility"""

    def __init__(self, level: LogLevel = LogLevel.INFO):
        """
        Initialize logger

        Args:
            level: Minimum log level to display
        """
        self.level = level

    def debug(self, message: str, *args: Any) -> None:
        """
        Log debug message

        Args:
            message: Log message
            *args: Additional arguments
        """
        self._log(LogLevel.DEBUG, message, *args)

    def info(self, message: str, *args: Any) -> None:
        """
        Log info message

        Args:
            message: Log message
            *args: Additional arguments
        """
        self._log(LogLevel.INFO, message, *args)

    def warn(self, message: str, *args: Any) -> None:
        """
        Log warning message

        Args:
            message: Log message
            *args: Additional arguments
        """
        self._log(LogLevel.WARN, message, *args)

    def error(self, message: str, *args: Any) -> None:
        """
        Log error message

        Args:
            message: Log message
            *args: Additional arguments
        """
        self._log(LogLevel.ERROR, message, *args)

    def set_level(self, level: LogLevel) -> None:
        """
        Set log level

        Args:
            level: New log level
        """
        self.level = level

    def get_level(self) -> LogLevel:
        """
        Get current log level

        Returns:
            Current log level
        """
        return self.level

    def _log(self, level: LogLevel, message: str, *args: Any) -> None:
        """
        Internal log method

        Args:
            level: Log level
            message: Log message
            *args: Additional arguments
        """
        if level.value >= self.level.value:
            timestamp = datetime.now().isoformat()
            formatted_message = f'[{timestamp}] [{level.name}] {message}'

            if args:
                formatted_message += f' {args}'

            print(formatted_message)

File: test-projects/javascript-sample/src/utils/Logger.js
/**
 * Simple logger utility
 */

export class Logger {
  constructor(level = 'info') {
    this.level = level;
    this.levels = {
      debug: 0,
      info: 1,
      warn: 2,
      error: 3
    };
  }

  /**
   * Log debug message
   * @param {string} message - Log message
   * @param {...any} args - Additional arguments
   */
  debug(message, ...args) {
    this.log('debug', message, ...args);
  }

  /**
   * Log info message
   * @param {string} message - Log message
   * @param {...any} args - Additional arguments
   */
  info(message, ...args) {
    this.log('info', message, ...args);
  }

  /**
   * Log warning message
   * @param {string} message - Log message
   * @param {...any} args - Additional arguments
   */
  warn(message, ...args) {
    this.log('warn', message, ...args);
  }

  /**
   * Log error message
   * @param {string} message - Log message
   * @param {...any} args - Additional arguments
   */
  error(message, ...args) {
    this.log('error', message, ...args);
  }

  /**
   * Set log level
   * @param {string} level - Log level
   */
  setLevel(level) {
    this.level = level;
  }

  /**
   * Get current log level
   * @returns {string} Current log level
   */
  getLevel() {
    return this.level;
  }

  /**
   * Internal log method
   * @param {string} level - Log level
   * @param {string} message - Log message
   * @param {...any} args - Additional arguments
   */
  log(level, message, ...args) {
    if (this.levels[level] >= this.levels[this.level]) {
      const timestamp = new Date().toISOString();
      const formattedMessage = `[${timestamp}] [${level.toUpperCase()}] ${message}`;

      switch (level) {
        case 'error':
          console.error(formattedMessage, ...args);
          break;
        case 'warn':
          console.warn(formattedMessage, ...args);
          break;
        default:
          console.log(formattedMessage, ...args);
      }
    }
  }
}
